<?xml version="1.0"?>

<launch>

    <arg name="model" default="$(find hector_quadrotor_description)/urdf/quadrotor.gazebo.xacro" />
    <arg name="world" default="generated" doc="launch specific or generated world"/>

    <!-- start own gazebo (test) world -->

    <include file="$(find hector_gazebo_worlds)/launch/start.launch">
        <arg name="world" value="$(find virtual_maize_field)/worlds/generated.world"/>
    </include>


    <!-- spawn UAVs -->
    <group ns="uav1">
        <include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor.launch">
            <arg name="name" value="uav1" />
            <arg name="tf_prefix" value="uav1" />
            <arg name="model" value="$(arg model)" />
            <arg name="x" value="5.0"/> 
            <arg name="y" value="0.0"/> 
            <arg name="z" value="0.3"/> 
        </include>

        <!-- launch hector node for uav1-->
        <node pkg="aro" type="hector_node.py" name="hector_node" output="screen" args="uav1">
            <remap from="/release" to="/uav1/release"/>
            <remap from="/ground_truth/state" to="/uav1/ground_truth/state"/>
            <remap from="/sonar_height" to="/uav1/sonar_height"/>
            <remap from="/set_corner" to="/uav1/set_corner"/>
            <remap from="/cmd_vel" to="/uav1/cmd_vel"/>
        </node>

        <!-- launch ds4_driver -->
        <include file="$(find ds4_driver)/launch/ds4_driver.launch" />

        <!-- run remote control node -->
        <node pkg="aro" type="status_to_hector.py" name="status_to_hector" output="screen">
            <rosparam command="load" file="$(find aro)/config/status_to_hector.yaml" />
            <remap from="/cmd_vel" to="/uav1/cmd_vel"/>
        </node>
    </group>

    <group ns="uav2">
        <include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor.launch">
            <arg name="name" value="uav2" />
            <arg name="tf_prefix" value="uav2" />
            <arg name="model" value="$(arg model)" />
            <arg name="x" value="5.0"/> 
            <arg name="y" value="2.0"/> 
            <arg name="z" value="0.3"/> 
        </include>
    </group>

</launch>